### ISSUE TITLE
[Feature Request]: Clean-up Astro Islands when removed from Document 

### ISSUE THREAD

**Issue by benjaminrobinet** [swup/astro#27]
### Describe the problem 🧐

While Swup is a good alternative to View Transitions in Astro, there is a big issue at the moment. (Hydrated) Astro Islands are not getting cleaned up, meaning it can leads to enormous memory leaks due to side effects.

### Describe the proposed solution 😎

Luckily, after looking a bit at how ViewTransitions works in Astro, I think there is a simple solution.

For now, that is how unmounting works in ViewTransitions:
When an Island is getting disconnected from the DOM, it register an event listener to `astro:after-swap`. The event listener will then trigger its own event: `astro:unmount`, which is listened in every astro integrations (vue, svelte, react, ...) to unmount itself.

Just some code to illustrate: 

AstroIsland disconnectedCallback: https://github.com/withastro/astro/blob/8e135b6f036a37c5df7884209e767e298262bad7/packages/astro/src/runtime/server/astro-island.ts#L54

View Transitions `astro:after-swap` event trigger: https://github.com/withastro/astro/blob/8e135b6f036a37c5df7884209e767e298262bad7/packages/astro/src/transitions/router.ts#L302

Vue integration unmounting through the `astro:unmount` event
https://github.com/withastro/astro/blob/8e135b6f036a37c5df7884209e767e298262bad7/packages/integrations/vue/client.js#L47

So, maybe I am missing some other side effects that needs to be cleaned up, but here is my simple fix for now:

```js
swup.hooks.on('content:replace', () => {
    triggerEvent(TRANSITION_AFTER_SWAP);
});
```

### Alternatives considered 🤔

Astro: View Transitions

### How important is this feature to you? 🧭

Critical

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for related problems
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by benjaminrobinet** [comment]
Looks like if the event is triggered before the customElement (Astro Island) got "disconnected" for the document, the event listener won't be attached. So yes I guess it matter.

**Comment by daun** [comment]
Works like a charm! I've added two other somewhat relevant events as well, for good measure. Feel free to test [the changes on the PR branch](https://github.com/swup/astro/pull/28), otherwise I guess we'll have this released in a day or two.

**Comment by daun** [comment]
Released in [1.5.0](https://github.com/swup/astro/releases/tag/1.5.0), thanks for the detailed write-up!
